worlds/hades/Rules.py

```diff
@@ -0,0 +1,421 @@
+                state._has_defeated_boss("LernieVictory", player, options))
+    
+    set_rule(world.get_location("PowerWithoutEqual"+subfix, player), lambda state: \
+                state._has_defeated_boss("BrosVictory", player, options))
+    add_rule(world.get_location("DivinePairings"+subfix, player), lambda state: \
+                state._has_defeated_boss("LernieVictory", player, options))
+        
+
+def set_weapon_region_rules(world: "HadesWorld", player: int, number_items: int, 
+                            location_table, options, weaponSubfix: str, total_routine_inspection:int):
+    set_rule(world.get_entrance("Zags room"+weaponSubfix, player), lambda state: \
+            state._has_weapon(weaponSubfix, player, options))
+    set_rule(world.get_entrance("Exit Tartarus"+weaponSubfix, player), lambda state: \
+            state.has("MegVictory"+weaponSubfix, player) and \
+            state._total_heat_level(player, min(number_items/4,10), options) and \
+            state._has_enough_routine_inspection(player,total_routine_inspection-2, options) and \
+            state._has_enough_weapons(player, options, 2))
+    set_rule(world.get_entrance("Exit Asphodel"+weaponSubfix, player), lambda state: \
+            state.has("LernieVictory"+weaponSubfix, player) and \
+            state._total_heat_level(player, min(number_items/2,20), options) and \
+            state._has_enough_routine_inspection(player,total_routine_inspection-1, options) and \
+            state._has_enough_weapons(player, options, 3))
+    set_rule(world.get_entrance("Exit Elyseum"+weaponSubfix, player), lambda state: \
+            state.has("BrosVictory"+weaponSubfix, player)  and \
+            state._total_heat_level(player, min(number_items*3/4,30), options) and \
+            state._has_enough_routine_inspection(player,total_routine_inspection, options) and \
+            state._has_enough_weapons(player, options, 5))
+    set_rule(world.get_location("Beat Hades"+weaponSubfix, player), lambda state: \
+            state._total_heat_level(player, min(number_items,35), options) and \
+            state._has_enough_weapons(player, options, 6))
+    
+
+def forbid_important_items_on_late_styx(world: "HadesWorld", player: int, options):
+    if options.location_system == "roomweaponbased":
+        for weaponString in location_weapons_subfixes:
+                late_styx_region = world.get_region("StyxLate"+weaponString, player)
+                for location in late_styx_region.locations:
+                        add_item_rule(location,
+                                lambda item: item_is_progression(item) == False or item_is_plando(world, item, player))
+    else:
+        late_styx_region = world.get_region("StyxLate", player)
+        for location in late_styx_region.locations:
+                add_item_rule(location,
+                        lambda item: item_is_progression(item) == False or item_is_plando(world, item, player))
```

Comment:
```suggestion
                        lambda item: not item.advancement or item_is_plando(world, item, player))
```
Found out about this one recently o: please test it before committing to make sure I'm not using it wrong!

Reply:
Tested it in python separately and on AP, and it works! Nice to know there is a shorthanded way of doing that.